api/app/Utilities/PostgresTextSearch.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Utilities;
+
+use Illuminate\Support\Str;
+
+enum PostgresTextSearchMatchingType
+{
+    case EXACT;
+    case PREFIX;
+}
+
+// https://www.postgresql.org/docs/current/datatype-textsearch.html#DATATYPE-TSQUERY
+// https://www.postgresql.org/docs/current/textsearch-controls.html#TEXTSEARCH-PARSING-QUERIES
+class PostgresTextSearch
+{
+    public static $invalidPatterns =
+        '/'.
+        '^&'.'|'.       // & at the beginning
+        '&$'.'|'.       // & at the end
+        '^\|'.'|'.      // | at the beginning
+        '\|$'.'|'.      // | at the end
+        ':'.'|'.        // : anywhere
+        '^.+?\!'.'|'.   // ! not at the beginning
+        '^\!$'.         // ! at the end
+        '/';
```

Comment:
Oh, seems like a few more characters to ignore to prevent errors. Unlikely to be used other than by accident, but prevents unneeded errors and logging. 
```
(
)
<
>
```

Reply:
I get errors with single quote alone and something starting like this  '[

Reply:
may be instead of adding invalid chars list , we could regex allowable chars and ignore others? upto you

Reply:
The big problem is that I don't know what the list of allowed or disallowed characters are.  I'd prefer to err on the side of permissiveness.  Let them try whatever they want.  The stakes are low if they try something that doesn't work.  They'll just get an error message.

Reply:
'[ is an error because it is essentially empty.  There's nothing to search for there.